src/main/java/lithium/university/services/TwitterService.java

```diff
@@ -31,57 +32,53 @@ private TwitterService() {}
      * Input: Twitter twitter - twitter instance, String message - message to be posted, int tweetTotal - total limited characters
      * Output: status object of updated status if successful update, null otherwise
      * */
-    public Status postToTwitter(Twitter twitter, String message, int tweetTotal) throws TwitterException, TwitterServiceException {
-        if (message == null) {
+    public Status postToTwitter(Twitter twitter, Optional<String> message, int tweetTotal) throws TwitterException, TwitterServiceException {
+        if(message.isPresent()) {
+            logger.info("Attempting to update status");
+            return Stream.of(twitter.updateStatus(message.filter(s -> s.length() > 0)
```

Comment:
does the s.length work or does an exception get thrown first?

Reply:
The segment of code will check that the message length is (0, 280] or else throws the exception. If you're referring to the potential null, the "message.isPresent()" check on line 36 takes care of that

Reply:
return optional and update syntax of stream